### PR TITLE
[Docker-Gen]  Implement more Dockerfile Commands

### PR DESCRIPTION
**Pull request by Lewiky**
Implement a load more Docker commands

- ENV
- ADD
- COPY
- ENTRYPOINT
- VOLUME
- WORKDIR

Unit tested

### CODE REVIEW

File: src/main/scala/temple/generate/docker/DockerfileGenerator.scala [hunk 1]
```diff
@@ -16,10 +16,16 @@ object DockerfileGenerator {
...
-      case Run(executable, params) => mkCode("RUN", buildArrayString(executable +: params))
-      case Cmd(executable, params) => mkCode("CMD", buildArrayString(executable +: params))
-      case Expose(port)            => mkCode("EXPOSE", port.toString)
+      case From(image, tag)               => mkCode("FROM", image, tag.map(":" + _))
+      case Run(executable, params)        => mkCode("RUN", buildArrayString(executable +: params))
+      case Cmd(executable, params)        => mkCode("CMD", buildArrayString(executable +: params))
+      case Expose(port)                   => mkCode("EXPOSE", port.toString)
+      case Env(key, value)                => mkCode("ENV", key, value)
+      case Add(src, dest)                 => mkCode("ADD", src, dest)
+      case Copy(src, dest)                => mkCode("COPY", src, dest)
```
Comment (c-harding) [inline src/main/scala/temple/generate/docker/DockerfileGenerator.scala:25]: Does this work with a space in the file path? Could that ever happen?

**Review by c-harding — COMMENTED** [review]
Looks good, just concerned about parsing errors

File: src/main/scala/temple/generate/docker/DockerfileGenerator.scala (hunk 1 above)
Comment (Lewiky) [reply src/main/scala/temple/generate/docker/DockerfileGenerator.scala:25]: Well to be a valid file path in the first place, any spaces would need to be escaped, which would get propagated through to the Dockerfile, which also supports escaped spaces in paths.

Comment (c-harding) [reply src/main/scala/temple/generate/docker/DockerfileGenerator.scala:25]: Not sure that’s true, but if you’re sure they’ve already been fully escaped then that’s fine

**Review by c-harding — APPROVED** [review]

**Review by jaylees14 — APPROVED** [review]

**Pull request merged**